internal/provider/resource_apply_module.go

```diff
@@ -545,13 +656,125 @@ func (t *applyModuleResource) getCurrentApplied(ctx context.Context,
 			if latestRun.IsDestroy && (latestRun.Status == sdktypes.RunApplied) && (latestRun.Apply != nil) {
 				moduleInfoOutput.wasSuccessfulDestroy = true
 			}
+
+			// Get the resolved variables from the run that produced the state.
+			resolvedVars, err := t.client.Run.GetRunVariables(ctx, &sdktypes.GetRunInput{ID: latestRun.Metadata.ID})
+			if err != nil {
+				diags.AddError("Failed to get resolved variables", err.Error())
+				return nil, diags
+			}
+
+			moduleInfoOutput.resolvedVariables = resolvedVars
 		} else {
 			// Current state has no run ID, so it must have been manually updated.
 			moduleInfoOutput.wasManualUpdate = true
 		}
+		return moduleInfoOutput, diags
 	}
 
-	return nil
+	return nil, diags
+}
+
+// extractRunError extracts the error from a run's logs (if the run errored out).
+func (t *applyModuleResource) extractRunError(ctx context.Context, run *sdktypes.Run) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	if run.Status == sdktypes.RunErrored {
+		var jobID string
+
+		switch {
+		case run.Apply != nil:
```

Comment:
The error could be in the plan stage even if the apply is not nil.

Reply:
I don't understand how that could be.  If the plan errored, would that not have prevented the apply being attempted?  More importantly at the moment, by what criteria should I determine whether the plan has the error?